src/gui/raster/qgscolorrampshaderwidget.cpp

```diff
@@ -388,7 +388,7 @@ void QgsColorRampShaderWidget::applyColorRamp()
 
       double value = currentItem->text( ValueColumn ).toDouble();
       double position = ( value - mMin ) / ( mMax - mMin );
-      currentItem->setData( ColorColumn, Qt::EditRole, ramp->color( position ) );
+      whileBlocking( static_cast<QgsTreeWidgetItemObject *>( currentItem ) )->setData( ColorColumn, Qt::EditRole, ramp->color( position ) );
```

Comment:
is `static_cast<QgsTreeWidgetItemObject *>` really needed here?

Reply:
yes, because `currentItem` is a `QTreeWidgetItem` that is not a `QObject` and do not provide signal/slots. Derived class `QgsTreeWidgetItemObject` is derived too from `QObject`.